pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -100,6 +112,73 @@ func (r *MachineSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // Reconcile reconciles CAPI and MAPI machines for their respective namespaces.
 func (r *MachineSyncReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx, "namespace", req.Namespace, "name", req.Name)
```

Comment:
Is req.Name/req.Namespace always for a MAPI Machine? Or does this also get triggered by reqs that contain a CAPI Machine?

Reply:
If this can come from either MAPI or CAPI Machines we need to change the namespace for getting the mapiMachine, otherwise if we use req.Namespace we can end up looking into the openshift-cluster-api ns for a MAPI Machine.

Reply:
Not 100% I'm following you here. 

> is req.Name/req.Namespace always for a MAPI Machine?

Nope, we can trigger reconciles from a CAPI machine or InfrastructureMachine we just rewrite the reconcile.Request to have the mapi namespace. 

https://github.com/openshift/cluster-capi-operator/blob/main/pkg/util/watch_filters.go#L41

Although, I agree - being explicit in looking in the MAPI namespace for the MAPI machine is less surprising :) Will update.

Reply:
Ah ok so we watch CAPI Machines but the req always corresponds to a MAPI Machine. req.Namespace is always the MAPI one then.
Ok cool, then either could've worked. NP

Reply:
Yeah, I think given it made you think twice keeping it as clear as possible is the way to go :)